crates/scheduler/src/call/crypto/hash.rs

```diff
@@ -83,9 +92,15 @@ fn update<B: Board>(mut call: SchedulerCall<B, api::update::Sig>) {
         let data = memory.get(*data, *length)?;
         match scheduler.applet.hashes.get_mut(*id as usize)? {
             #[cfg(feature = "board-api-crypto-sha256")]
-            HashContext::Sha256(context) => context.update(data),
+            HashContext::Sha256(context) => {
+                context.update(data);
+                context.last_error()?
+            }
             #[cfg(feature = "board-api-crypto-sha384")]
-            HashContext::Sha384(context) => context.update(data),
+            HashContext::Sha384(context) => {
+                context.update(data);
+                context.last_error()?
```

Comment:
We don't want to "unwrap" the error. Within the try-block, errors are only for traps. Here, we want to return the error to the user.
```suggestion
                context.last_error()
```